provider/agent.go

```diff
@@ -259,6 +259,66 @@ func agentResource() *schema.Resource {
 				ForceNew:    true,
 				Optional:    true,
 			},
+			"resources_monitoring": {
+				Type:        schema.TypeSet,
+				Description: "The resources monitoring configuration for this agent.",
+				ForceNew:    true,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"memory": {
+							Type:        schema.TypeSet,
+							Description: "The memory monitoring configuration for this agent.",
+							ForceNew:    true,
+							Optional:    true,
```

Comment:
Consider adding `MinItems` and `MaxItems` for these resources.

Reply:
+1

Reply:
I added a `MaxItems` for memory to avoid duplicates. 👌